manifests/config/global/jenkins_config.pp

```diff
@@ -0,0 +1,24 @@
+# This private class is called from jenkins::config to manage Jenkins' config.xml
+#
+class jenkins::config::global::jenkins_config(
+  $localstatedir           = $::jenkins::localstatedir,
+  $user                    = $::jenkins::user,
+  $group                   = $::jenkins::group,
+  $jenkins_config_template = $::jenkins::jenkins_config_template,
+) {
+  if $caller_module_name != $module_name {
+    fail("Use of private class ${name} by ${caller_module_name}")
+  }
+
+  $config_file = "${localstatedir}/config.xml"
+
+  file { $config_file:
+    ensure  => present,
+    content => template($jenkins_config_template),
+    owner   => $user,
+    group   => $group,
+    mode    => '0644',
+    require => File[$localstatedir],
+    notify  => Service[$::jenkins::params::service_name],
```

Comment:
I'm not sure but there could be an ordering issue here if `config.xml` is overwritten by the jenkins package.